### PR TITLE
feat(tcp-outlet): tuitify and make `ockam tcp-outlet show` (no args) interactive 

### PR DESCRIPTION
**Pull request by Lumexralph**
## Current behavior

As described in this issue #6477

## Proposed changes

The `ockam tcp-outlet show` is made to be interactive as suggested in this issue #6477

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] There are no Merge commits in this Pull Request. Ockam repo maintains a linear commit history. We merge Pull Requests by rebasing them onto the develop branch. Rebasing to the latest develop branch and force pushing to your Pull Request branch is okay.
- [x] I have read and accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have read and accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/.github/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam/blob/develop/.github/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam](https://github.com/build-trust/ockam) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam/edit/develop/.github/CONTRIBUTORS.csv) file in the github web UI and create a separate Pull Request, this will mark the commit as verified.

### CODE REVIEW

File: implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs [hunk 1]
```diff
@@ -67,28 +73,142 @@ pub async fn run_impl(
...
+    aliases_node: HashMap<String, BackgroundNode>,
+}
+
+impl ShowTui {
+    pub async fn run(
+        ctx: Context,
+        opts: CommandGlobalOpts,
+        cmd: ShowCommand,
+    ) -> miette::Result<()> {
+        let mut running_node_names: Vec<String> = Vec::new();
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs]: Note that you only have to show the TCP outlets of the given node at `node_opts.at_node` or default node if none is provided.

File: implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs [hunk 2]
```diff
@@ -67,28 +73,142 @@ pub async fn run_impl(
...
+                .filter(|node| node.is_running())
+                .filter_map(|node| extract_address_value(node.name()).ok())
+                .collect();
+        }
+
+        if running_node_names.is_empty() {
+            return Err(miette!("No running nodes found"));
+        }
+
+        let mut outlet_aliases: Vec<String> = Vec::new();
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs]: The logic of gathering the outlets names should be behind the trait's `list_items_names(&self)` method.

File: implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs [hunk 3]
```diff
@@ -67,28 +73,142 @@ pub async fn run_impl(
...
+            let res = send_request(&ctx, &opts, node_name.clone()).await;
+            match res {
+                Ok(outlets) => {
+                    let mut node_outlets: Vec<String> = outlets
+                        .list
+                        .iter()
+                        .map(|outlet| outlet.alias.clone())
+                        .collect();
+
+                    let node = BackgroundNode::create(&ctx, &opts.state, &node_name).await?;
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs]: You can store this in the struct as it's a constant once you know what node you want to query. Here is an [example](https://github.com/build-trust/ockam/blob/54bb211c0f1eb20af324cc5e37b38f3a43ff9817/implementations/rust/ockam/ockam_command/src/relay/show.rs#L56) of how we did it for another command.

File: implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs [hunk 4]
```diff
@@ -67,28 +73,142 @@ pub async fn run_impl(
...
+            .alias
+            .clone()
+            .ok_or(miette!("No TCP Outlet alias provided"))
+    }
+
+    async fn list_items_names(&self) -> miette::Result<Vec<String>> {
+        Ok(self.outlet_aliases.clone())
+    }
+
+    async fn show_single(&self, item_name: &str) -> miette::Result<()> {
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs:141]: This function could be simplified after you apply the changes from the other comments.

**Review by adrianbenavides — COMMENTED** [review]
Hey @Lumexralph, thanks for the PR! I've left some comments, let me know if you have any questions 👍

File: implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs (hunk 1 above)
Comment (Lumexralph) [reply implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs]: Ha! Gotcha.

I apologize for the late response on this.

File: implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs (hunk 4 above)
Comment (Lumexralph) [reply implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs:141]: cool! Thanks for the feedback

File: implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs (hunk 1 above)
Comment (adrianbenavides) [reply implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs]: No worries! Take the time you need 🙏

Comment (Lumexralph) [reply implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs]: @adrianbenavides  feedback implemented

**Comment by adrianbenavides** [comment]
Hey @Lumexralph, your PR will be landed through https://github.com/build-trust/ockam/pull/7309. I've added you as co-author, so you keep the authorship of the changes. Thanks again for your contribution! 🎉

**Pull request closed without merging**